packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -470,6 +480,13 @@ export const useField = <
     setSelectedSections: (activeSectionIndex) => setSelectedSections(activeSectionIndex),
   }));
 
+  const handleClearValue = (...args) => {
+    onClear?.(...(args as []));
+    clearValue();
+    inputRef?.current?.focus();
+    setSelectedSections(0);
```

Comment:
This would need a deeper investigation in order to uncover the root cause of the issue, but it shouldn't be difficult tracking the stack trace of each of the `setSelectedSections` calls. 🤔